### PR TITLE
JavaScript2 Week1 Homework

### PR DESCRIPTION
**Pull request by ihk2018**

### CODE REVIEW

File: Week1/homework/app.js [hunk 1]
```diff
@@ -1,11 +1,161 @@
...
+    in_search_of_lost_time: './img/in_search_of_lost_time.jpg',
+    great_expectations: ' ./img/great_expectations.jpg',
+    war_and_peace: ' ./img/war_and_peace.jpg',
+    the_great_gatsby: ' ./img/the_great_gatsby.jpg',
+    to_kill_a_mockingbird: ' ./img/to_kill_a_mockingbird.jpg',
+    the_adventures_of_huckleberry_finn: ' ./img/the_adventures_of_huckleberry_finn.jpg',
+  };
+
+
+  const newDiv = document.createElement('div');
```
Comment (J05J4) [inline Week1/homework/app.js]: How "new" is this `<div>`? New div isn't the best name for this element and variable. Maybe you could use a semantic element and give it a more descriptive name. Same goes for `newH1`, `newUl`, `newImg` etc.

File: Week1/homework/app.js [hunk 2]
```diff
@@ -1,11 +1,161 @@
...
+      const newLi = document.createElement('li');
+      newUl.appendChild(newLi);
+      const newImg = document.createElement('img');
+      newLi.appendChild(newImg);
+      newImg.src = Object.values(bookCovers)[i];
+      const newHeadingH2 = document.createElement('h2');
+      const newHeadingH3 = document.createElement('h3');
+      newLi.appendChild(newHeadingH2);
+      newLi.appendChild(newHeadingH3);
+      const paraGenre = document.createElement('p');
```
Comment (J05J4) [inline Week1/homework/app.js]: What is `paraGenre` or `paraLanguage`? You should use descriptive names.

File: Week1/homework/app.js [hunk 3]
```diff
@@ -1,11 +1,161 @@
...
+    war_and_peace: ' ./img/war_and_peace.jpg',
+    the_great_gatsby: ' ./img/the_great_gatsby.jpg',
+    to_kill_a_mockingbird: ' ./img/to_kill_a_mockingbird.jpg',
+    the_adventures_of_huckleberry_finn: ' ./img/the_adventures_of_huckleberry_finn.jpg',
+  };
+
+
+  const newDiv = document.createElement('div');
+  console.log(newDiv);
+  document.getElementsByTagName('body')[0].appendChild(newDiv);
```
Comment (J05J4) [inline Week1/homework/app.js]:
```suggestion
  document.body.appendChild(newDiv);
```

File: Week1/homework/app.js [hunk 4]
```diff
@@ -1,11 +1,161 @@
...
+      const paraCountry = document.createElement('p');
+      newLi.appendChild(paraCountry);
+      const detailsOfBooksItem = detailsOfBooks[i];
+      newHeadingH2.innerText = detailsOfBooksItem.title;
+      newHeadingH3.innerText = detailsOfBooksItem.author;
+      paraGenre.innerText = detailsOfBooksItem.genre;
+      paraLanguage.innerText = detailsOfBooksItem.language;
+      paraCountry.innerText = detailsOfBooksItem.country;
+      console.log(newHeadingH2.innerText);
+      newImg.className = 'images';
```
Comment (J05J4) [inline Week1/homework/app.js]: Your `img` has now `alt` attribute.

File: Week1/homework/app.js [hunk 5]
```diff
@@ -1,11 +1,161 @@
...
+
+    const detailsOfBooks = Object.values(infoForEachBook);
+    console.log(detailsOfBooks);
+
+    for (let i = 0; i < detailsOfBooks.length; i++) {
+      const newLi = document.createElement('li');
+      newUl.appendChild(newLi);
+      const newImg = document.createElement('img');
+      newLi.appendChild(newImg);
+      newImg.src = Object.values(bookCovers)[i];
```
Comment (J05J4) [inline Week1/homework/app.js]: On line 121 you are generating an array containing the values of the infoForEachBook:
```
    const detailsOfBooks = Object.values(infoForEachBook);
```
On line 129 you are doing something similar for the object of bookCovers. 
```
      newImg.src = Object.values(bookCovers)[i];
```
You are assuming the order of those arrays is the same. The homework says: "Remember that objects are not ordered, so you cannot guarantee that the first key is the first li element. (Hint: you could give each li item an id tag by modifying the function you made before.)"
Other than that you are now using those objects as if they where arrays and you are not at all using `bookTitles` anymore. 
The idea in this homework is to have an array with book titles, and two objects with information. You should loop over the array of book titles and look for the key in both objects to add the right information.

File: Week1/homework/app.js [hunk 6]
```diff
@@ -1,11 +1,161 @@
...
+
+
+  function generateMyFavoriteBookList() {
+
+    const newUl = document.createElement('ul');
+    console.log(newUl);
+    newDiv.appendChild(newUl);
+
+    const detailsOfBooks = Object.values(infoForEachBook);
+    console.log(detailsOfBooks);
```
[the remark this answers is not in the document]
Comment (ihk2018) [reply Week1/homework/app.js]: Dear Josja, thank you for this detailed and explanatory feedback, I am working on it to improve.

Regards,

File: Week1/homework/app.js (hunk 1 above)
Comment (ihk2018) [reply Week1/homework/app.js]: I changed all the names starting with “new”. My intention was to show that I created new elements but it was not a good idea.

**Comment by Tjebbee** [comment]
@ihk2018 , Are you done implementing the feedback from @J05J4 ? If so, please contact him to check your changes! :)

**Comment by ihk2018** [comment]
> @ihk2018 , Are you done implementing the feedback from @J05J4 ? If so, please contact him to check your changes! :)

Hi @Tjebbee ,thank you for reminding me. I implemented the needed changes last week, and today I requested from @J05J4  to review it again. Regards,

**Pull request closed without merging**